qa/integration-tests/src/test/java/io/camunda/zeebe/it/clustering/AsymmetricNetworkPartitionIT.java

```diff
@@ -0,0 +1,287 @@
+
+    // when
+    final var future = when.apply(zeebeClient);
+    removeAsymmetricNetworkPartition(
+        ipAddress, getContainerForNodeId(leaderOfPartitionThree.getNodeId()));
+
+    // then
+    then.accept(zeebeClient, future);
+  }
+
+  private BrokerInfo getPartitionLeader(final Topology topology, final int partition) {
+    return topology.getBrokers().stream()
+        .filter(
+            b ->
+                b.getPartitions().stream()
+                    .filter(p -> p.getPartitionId() == partition)
+                    .anyMatch(PartitionInfo::isLeader))
+        .findFirst()
+        .orElseThrow();
+  }
+
+  private void removeAsymmetricNetworkPartition(
+      final String ipAddress, final ZeebeBrokerNode<? extends GenericContainer<?>> brokerNode)
+      throws IOException, InterruptedException {
+    LOGGER.info(
+        "{}",
+        runCommandInContainer(brokerNode, "ip route del unreachable " + ipAddress).getStdout());
+  }
+
+  /**
+   * Set the given ip address unreachable for the given container.
+   *
+   * @param ipAddress the ip address which should be unreachable
+   * @param brokerNode the broker container which should be updated
+   * @throws IOException Can be thrown during running commands in the container
+   * @throws InterruptedException Can be thrown during running commands in the container
+   */
+  private void setupAsymmetricNetworkPartition(
+      final String ipAddress, final ZeebeBrokerNode<? extends GenericContainer<?>> brokerNode)
+      throws IOException, InterruptedException {
+    runCommandInContainer(brokerNode, "apt update");
+    runCommandInContainer(brokerNode, "apt install -y iproute2");
+    LOGGER.info(
+        "{}",
+        runCommandInContainer(brokerNode, "ip route add unreachable " + ipAddress).getStdout());
+  }
+
+  private ZeebeBrokerNode<? extends GenericContainer<?>> getContainerForNodeId(final int nodeId) {
+    return cluster.getBrokers().get(nodeId);
+  }
+
+  private String getContainerIpAddress(
+      final ZeebeBrokerNode<? extends GenericContainer<?>> leaderOneNode) {
+    return leaderOneNode
+        .getCurrentContainerInfo()
+        .getNetworkSettings()
+        .getNetworks()
+        .values()
+        .stream()
+        .findFirst()
+        .orElseThrow()
+        .getIpAddress();
+  }
+
+  private ExecResult runCommandInContainer(
+      final ZeebeBrokerNode<? extends GenericContainer<?>> container, final String command)
+      throws IOException, InterruptedException {
+    LOGGER.info("Run command: {}", command);
+
+    final var commands = command.split(" ");
+    final var execResult = container.execInContainer(commands);
+
+    if (execResult.getExitCode() == 0) {
+      LOGGER.info("Command {} was successful.", command);
+    } else {
+      LOGGER.error("Command {} failed with code: {}", command, execResult.getExitCode());
+      LOGGER.error("Stderr: {}", execResult.getStderr());
```

Comment:
:wrench: I think we should throw here, otherwise I'd imagine failing tests are difficult to debug when, for example, `apt update` fails.